### DISCUSSION TITLE
建议增加停用guest账号的功能

### DISCUSSION THREAD

**Discussion by mingbai2006** [AlistGo/alist#3018] [Ideas]
现在的guest账号即使什么权限都不选，也能去访问“基本路径”下的内容，但有些用户只是想让NAS的目录能外网让自己访问，并不想开启guest用户，会有隐私问题。我现在只能把guest的“基本路径”设定为一个空目录来规避隐私问题，但还是希望系统层面能加一个禁用guest账号的功能

**Comment by xzxiaoshan** [comment]
不能直接删除吗？

**Reply by mingbai2006** [reply]
不允许删除，提示“Cannot delete admin or guest”

**Reply by shuanglu1** [reply]
你可以弄元信息加密啊，这样guest不就访问不了！

**Reply by mingbai2006** [reply]
你说的元信息方案还不如我po的方案，原因如下：
1、元信息需要逐个对目录起作用，不如禁用guest对整个站点都起作用这么方便
2、元信息会误伤没有开启‘’无需密码“的账号，比如你还有个别的普通账号，没有开启免密码访问的功能，他们在访问带有元信息加密的目录时就会被要求输密码
但无论我的方案还是元信息方案，都不如直接系统提供一个禁用guest账号的方案简单直观

**Comment by oneisall8955** [comment]
确实，我当前的的解决办法有点low，postman去请求接口，修改guest的用户路径为一个无效的路径。打开主页，抛出异常
![image](https://user-images.githubusercontent.com/44989283/212380267-e5ae3dff-c1f9-43ab-8d9c-c50eb058dada.png)

**Reply by mingbai2006** [reply]
我现在的做法跟你一样，属于是变相的解决了，但很不优雅

**Comment by jemiryguo** [comment]
我有一个稍微好一点的解决方法，就是在某个网盘上新建一个LittleBlackRoom的空文件夹，然后再用户管理里面设置guest用户的基本路径为这个小黑屋，这样游客访问就只能在这个小黑屋文件夹里面。还可以放些txt，写点公告什么的

**Comment by fengchen-github** [comment]
guest确实鸡肋啊

**Reply by fengchen-github** [reply]
我的做法是建一个空文件夹给guest作为根目录

**Reply by alanhe421** [reply]
> 我的做法是建一个空文件夹给guest作为根目录

当前唯一但不优雅的办法，比如自己代理层面判断下状态也行，但更麻烦了

**Comment by evlic** [comment]
guest 权限默认是……全部关闭的吧，应该什么都看不见

**Comment by evlic** [comment]
guest 权限默认是……全部关闭的吧，应该什么都看不见

**Comment by ErDongChen1973** [comment]
> 现在的guest账号即使什么权限都不选，也能去访问“基本路径”下的内容，但有些用户只是想让NAS的目录能外网让自己访问，并不想开启guest用户，会有隐私问题。我现在只能把guest的“基本路径”设定为一个空目录来规避隐私问题，但还是希望系统层面能加一个禁用guest账号的功能

可以在元信息里面给整个根目录设置一个密码，这样guest也看不到了

**Comment by xhofe** [comment]
available in v3.10.1
